src/builtin.h

```diff
@@ -44,7 +44,10 @@ struct builtin_data_t {
 #define BUILTIN_ERR_COMBO2_EXCLUSIVE _(L"%ls: %ls %ls: options cannot be used together\n")
 
 /// Error message on multiple scope levels for variables.
-#define BUILTIN_ERR_GLOCAL _(L"%ls: scope can be only one of: universal function global local\n")
+#define BUILTIN_ERR_GLOCAL _(L"%ls: scope can be only one of function, global, local\n")
```

Comment:
Maybe one way forward is to get rid of automatic persistence but make it easy to write the startup file.

A new `set --broadcast` command could eventually replace `--universal`.
`set --broadcast` uses a similar propagation logic to `--universal` but always interacts with globals, never universals.

```
$ set --broadcast fish_color_command green
Updating $fish_color_command in all running shells. 
Use this value for future shells in your startup file? [Ny]
y
echo 'set --global fish_color_command green' >>~/.config/fish/config.fish
```

This will eliminate the need for fish_user_paths:

```
$ set --broadcast --prepend PATH /opt/mycoolthing
Updating $PATH in all running shells. 
Use this value for future shells in your startup file? [Ny]
y
echo 'set --global --export --prepend PATH /opt/mycoolthing' >>~/.config/fish/config.fish
```

Observe that the exported bit is preserved.

Unfortunately this also makes it easy to add too much information to config.fish:

```
$ set --broadcast PATH /opt/mycoolthing $PATH
Updating $PATH in all running shells. 
Use this value for future shells in your startup file? [Ny]
y
echo 'set --global --export PATH /opt/mycoolthing /home/user/.local/bin /usr/local/bin /usr/bin' >>~/.config/fish/config.fish
```

Also the config file can get stuffed when using this a lot (though that's fairly easy to fix).

One advantage of this interactive command is that it avoids ever-growing variables in startup files like `set -Ua fish_user_paths /opt/mycoolthing` (although we have fixed that particular case by deduplicating `fish_user_paths`)

Sadly I don't have an idea of what to do with scripts that run `set -U` (non-interactively) and having both solutions around is not nice.